src/secp256k1.c

```diff
@@ -144,8 +146,12 @@ secp256k1_context* secp256k1_context_preallocated_create(void* prealloc, unsigne
 }
 
 secp256k1_context* secp256k1_context_create(unsigned int flags) {
-    size_t const prealloc_size = secp256k1_context_preallocated_size(flags);
-    secp256k1_context* ctx = (secp256k1_context*)checked_malloc(&default_error_callback, prealloc_size);
+    size_t prealloc_size;
+    secp256k1_context* ctx;
+
+    CHECK(secp256k1_selftest());
+    prealloc_size = secp256k1_context_preallocated_size(flags);
+    ctx = (secp256k1_context*)checked_malloc(&default_error_callback, prealloc_size);
```

Comment:
I think this is what we should do, namely call `default_error_callback`, which can at least be overridden at compile time.  We do this here because we can't call the error callback unfortunately. It's just not yet determined at context creation time. I'm going to add a note to #780, maybe we can improve this in the future.